packages/core/src/main/mls/MLSService/MLSService.ts

```diff
@@ -0,0 +1,177 @@
+  RemoveProposalArgs,
+} from '@otak/core-crypto/platforms/web/corecrypto';
+import {APIClient} from '@wireapp/api-client';
+import {QualifiedUsers} from '../../conversation';
+import {Decoder, Encoder} from 'bazinga64';
+import type {MLSConfig} from '../types';
+
+//@todo: this function is temporary, we wait for the update from core-crypto side
+//they are returning regular array instead of Uint8Array for commit and welcome messages
+export const optionalToUint8Array = (array: Uint8Array | []): Uint8Array => {
+  return Array.isArray(array) ? Uint8Array.from(array) : array;
+};
+
+export class MLSService {
+  constructor(
+    public readonly config: MLSConfig | undefined,
+    private readonly apiClient: APIClient,
+    private readonly coreCryptoClientProvider: () => CoreCrypto | undefined,
+  ) {}
+
+  private getCoreCryptoClient() {
+    const client = this.coreCryptoClientProvider();
+    if (!client) {
+      throw new Error('Could not get coreCryptoClient');
+    }
+    return client;
+  }
+
+  public async uploadCoreCryptoCommitBundle(groupIdDecodedFromBase64: Uint8Array, commitBundle: CommitBundle) {
+    const coreCryptoClient = this.getCoreCryptoClient();
+
+    if (commitBundle.welcome) {
+      //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
+      await this.apiClient.api.conversation.postMlsWelcomeMessage(optionalToUint8Array(commitBundle.welcome));
+    }
+    if (commitBundle.commit) {
+      const messageResponse = await this.apiClient.api.conversation.postMlsMessage(
+        //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
+        optionalToUint8Array(commitBundle.commit),
+      );
+      await coreCryptoClient.commitAccepted(groupIdDecodedFromBase64);
+      return messageResponse;
+    }
+    return null;
+  }
+
+  public async addUsersToExistingMLSConversation(groupIdDecodedFromBase64: Uint8Array, invitee: Invitee[]) {
+    const coreCryptoClient = this.getCoreCryptoClient();
+    const memberAddedMessages = await coreCryptoClient.addClientsToConversation(groupIdDecodedFromBase64, invitee);
+
+    if (memberAddedMessages?.welcome) {
+      //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
+      await this.apiClient.api.conversation.postMlsWelcomeMessage(optionalToUint8Array(memberAddedMessages.welcome));
+    }
+    if (memberAddedMessages?.commit) {
+      const messageResponse = await this.apiClient.api.conversation.postMlsMessage(
+        //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
+        optionalToUint8Array(memberAddedMessages.commit),
+      );
+      await coreCryptoClient.commitAccepted(groupIdDecodedFromBase64);
+      return messageResponse;
+    }
+    return null;
+  }
+
+  public async getCoreCryptoKeyPackagesPayload(qualifiedUsers: QualifiedUsers[]) {
```

Comment:
```suggestion
  public async getKeyPackagesPayload(qualifiedUsers: QualifiedUsers[]) {
```

No need to mention CoreCrypto here